types/time.go

```diff
@@ -1820,10 +1829,115 @@ func ExtractTimeValue(unit string, format string) (int64, int64, int64, float64,
+	switch strings.ToUpper(unit) {
+	case "MICROSECOND", "SECOND", "MINUTE", "HOUR", "DAY", "WEEK", "MONTH", "QUARTER", "YEAR":
+		return extractSingleDurationValue(unit, format)
+	case "SECOND_MICROSECOND":
+		_, _, _, d, err := extractSecondMicrosecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(d), Fsp: MaxFsp}, nil
+	case "MINUTE_MICROSECOND":
+		_, _, _, d, err := extractMinuteMicrosecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(d), Fsp: MaxFsp}, nil
+	case "MINUTE_SECOND":
+		_, _, _, d, err := extractMinuteSecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(d), Fsp: MaxFsp}, nil
+	case "HOUR_MICROSECOND":
+		_, _, _, d, err := extractHourMicrosecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(d), Fsp: MaxFsp}, nil
+	case "HOUR_SECOND":
+		_, _, _, d, err := extractHourSecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(d), Fsp: MaxFsp}, nil
+	case "HOUR_MINUTE":
+		_, _, _, d, err := extractHourMinute(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(d), Fsp: 0}, nil
+	case "DAY_MICROSECOND":
+		_, _, day, dur, err := extractDayMicrosecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(day*int64(GoDurationDay) + int64(dur)), Fsp: MaxFsp}, nil
+	case "DAY_SECOND":
+		_, _, day, dur, err := extractDaySecond(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(day*int64(GoDurationDay) + int64(dur)), Fsp: MaxFsp}, nil
+	case "DAY_MINUTE":
+		_, _, day, dur, err := extractDayMinute(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(day*int64(GoDurationDay) + int64(dur)), Fsp: 0}, nil
+	case "DAY_HOUR":
+		_, _, day, dur, err := extractDayHour(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		return Duration{Duration: gotime.Duration(day*int64(GoDurationDay) + int64(dur)), Fsp: 0}, nil
+	case "YEAR_MONTH":
+		_, _, _, _, err := extractYearMonth(format)
+		if err != nil {
+			return ZeroDuration, err
+		}
+		// MONTH must exceed the limit of mysql's duration. So just return overflow error.
+		return ZeroDuration, ErrDatetimeFunctionOverflow.GenWithStackByArgs("time")
+	default:
+		return ZeroDuration, errors.Errorf("invalid single timeunit - %s", unit)
 	}
 }
 
+func extractSingleDurationValue(unit string, format string) (Duration, error) {
+	fv, err := strconv.ParseFloat(format, 64)
+	if err != nil {
+		return ZeroDuration, ErrIncorrectDatetimeValue.GenWithStackByArgs(format)
+	}
+	iv := int64(math.Round(fv))
+
+	switch strings.ToUpper(unit) {
```

Comment:
Redundant computing as `ExtractDurationValue` already call the `strings.ToUpper` function